src/App.tsx

```diff
@@ -15,48 +17,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  string = 'alphabetically',
+  num = 'length',
+}
+
+interface GoodsProps {
+  sortField: SortField | '';
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: GoodsProps,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.string:
+          return good1.localeCompare(good2);
+
+        case SortField.num:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
```

Comment:
Use explicit initial state for 'sortField'. For example, you could define 'SortField.NONE' and use it here.